### PR TITLE
[KL-423] Navigate to requested page when Maxview autoopens

### PR DESCRIPTION
**Pull request by thesvyatoslavtt**

### CODE REVIEW

File: libs/ui/src/utils/redirecit-to-maximise-view.util.ts [hunk 1]
```diff
@@ -1,9 +1,13 @@
+import { OnEventFn } from '@rnw-community/shared';
+
 import { isFullpage } from './location.utils';
-import { openMaximiseScreen } from './open-maximise-screen.util';
+import { openFullViewPage } from './open-maximise-screen.util';
 import { isWeb } from './platform.utils';
 
-export const redirectToMamixiseView = () => {
+export const redirectToFullViewPage = (fn: OnEventFn<void>) => {
+  fn();
```
Comment (MuKoJla) [inline libs/ui/src/utils/redirecit-to-maximise-view.util.ts]: maybe we can use here `globalNavigationRef` and `redirectToFullViewPage` can accept as arg just screenEnum?

File: libs/ui/src/components/navigator/navigator.tsx [hunk 2]
```diff
@@ -88,16 +88,43 @@ export const Navigator: FC = () => {
...
-    // TODO: Add check for ScreenEnum.AlmostDone screen later
-    const isCreateWalletScreensOpened =
+    const IsOpenOnlyFullViewScreen =
       initialState?.routes.some(
-        route => route.name === ScreensEnum.CreateANewWallet || route.name === ScreensEnum.VerifyMnemonic
+        route =>
+          route.name === ScreensEnum.AddNewCollectible ||
+          route.name === ScreensEnum.AddNewToken ||
+          route.name === ScreensEnum.AddNetwork
       ) ?? false;
```
Comment (Boltman92) [inline libs/ui/src/components/navigator/navigator.tsx:97]: Array.some() return boolean, why we add `?? false` ?

File: libs/ui/src/utils/redirecit-to-maximise-view.util.ts [hunk 3]
```diff
@@ -1,9 +1,13 @@
...
+
   if (isWeb && !isFullpage) {
-    openMaximiseScreen();
+    setTimeout(openFullViewPage, 0);
```
Comment (Boltman92) [inline libs/ui/src/utils/redirecit-to-maximise-view.util.ts]: you could skip 0 if you don't need to specify delay, leave just function inside setTimeout

**Review by MuKoJla — DISMISSED** [review]

File: libs/ui/src/components/navigator/navigator.tsx [hunk 4]
```diff
@@ -88,16 +88,43 @@ export const Navigator: FC = () => {
...
+        JSON.stringify({
+          ...initialState,
+          routes: [
+            { key: 'Welcome-YOKi4uV67S17eluroXFWt', name: 'Welcome' },
+            { key: `${parentRoute}-h-XhNwryARoSQpUS0E3f-`, name: parentRoute }
+          ]
+        })
+      );
+
+      openFullViewPage();
```
Comment (unixvb) [inline libs/ui/src/components/navigator/navigator.tsx]: This hook looks complicated and non-explicit: hard conditions, auto-generated keys `YOKi4uV67S17eluroXFWt`, `h-XhNwryARoSQpUS0E3f`, changing store state, etc.  

Rewrite this hook and `redirectToFullViewPage` function.  
You could make some query params (for example `initialScreen: ScreensEnum`) and if that value defined - push that route into `storedInitialState` inside `usePersistedNavigationState` hook

**Review by unixvb — APPROVED** [review]

**Review by MuKoJla — APPROVED** [review]

**Review by Boltman92 — APPROVED** [review]

**Review by vladvn1905 — APPROVED** [review]

**Pull request merged**